tests: reading outside array bounds in iconvnm.c

The good_names array in iconvnm.c is empty for me which gives a warning when compiling the file:
```
../picolibc/newlib/testsuite/newlib.iconv/iconvnm.c:91:35: warning: array subscript 0 is above array bounds of 'char *[0]' [-Warray-bounds]
   91 |         cd = iconv_open(good_names[0], bad_names[i]);
      |                         ~~~~~~~~~~^~~
```

I'm guessing this test (and the iconvjp/iconvru tests) should not be enabled when`-Dnewlib-iconv-encodings=none`.

--- a/newlib/testsuite/meson.build
+++ b/newlib/testsuite/meson.build
@@ -33,8 +33,13 @@
 # OF THE POSSIBILITY OF SUCH DAMAGE.
 #
 
-dirs = [
-	'newlib.iconv',
+if newlib_iconv_encodings == []
+  dirs = []
+else
+  dirs = ['newlib.iconv']
+endif
+
+dirs += [
 	'newlib.locale',
 	'newlib.search',
 	'newlib.stdio',

